src/webgpu/web_platform/copyToTexture/canvas.spec.ts

```diff
@@ -51,53 +38,137 @@ class F extends CopyToTextureUtils {
       unreachable();
     }
 
+    return canvas;
+  }
+
+  init2DCanvasContent({
+    canvasType,
+    width,
+    height,
+    isLosePrecisionDstFormat,
+  }: {
+    canvasType: 'onscreen' | 'offscreen';
+    width: number;
+    height: number;
+    isLosePrecisionDstFormat: boolean;
+  }): {
+    canvas: HTMLCanvasElement | OffscreenCanvas;
+    canvasContext: CanvasRenderingContext2D | OffscreenCanvasRenderingContext2D;
+  } {
+    const canvas = this.createCanvas(canvasType, width, height);
     if (canvas === null) {
       this.skip('Cannot create canvas');
     }
 
-    const canvasContext = canvas.getContext(contextName) as
-      | WebGLRenderingContext
-      | WebGL2RenderingContext
+    let canvasContext = null;
+    canvasContext = canvas.getContext('2d') as
       | CanvasRenderingContext2D
       | OffscreenCanvasRenderingContext2D
       | null;
+
     if (canvasContext === null) {
       this.skip(canvasType + ' canvas context not available');
```

Comment:
```suggestion
      this.skip(canvasContext + ' canvas context not available');
```

Reply:
I expect the error message should be : "onscreen canvas : canvas context not available". So wants the keep 'canvasType' here.

Reply:
Availability of the canvasType was already checked, so maybe here it makes sense to report both. So e.g. if offscreen + 2d is not available, it says that.

Though it doesn't really matter I suppose, the canvas type and context type will already be in the test case name!

Reply:
<  the canvas type and context type will already be in the test case name!
Good point!